modules/fedimint-lnv2-server/src/lib.rs

```diff
@@ -333,6 +337,16 @@ impl ServerModule for Lightning {
         &self,
         _dbtx: &mut DatabaseTransaction<'_>,
     ) -> Vec<LightningConsensusItem> {
+        if tokio::time::timeout(Duration::from_secs(15), self.propose_citem.notified())
+            .await
+            .is_err()
+        {
+            // Just update time
+            return vec![LightningConsensusItem::UnixTimeVote(
+                duration_since_epoch().as_secs(),
+            )];
+        }
+
```

Comment:
you will miss notification here, because you only start listening from when this function is called this time till timeout.  What you want is listen for notification since *last* call to this function till timeout. (notification between last call and current call are missed due to how notify works)

Reply:
@maan2003 Nooooo? I either misunderstand you or docs or you misunderstand something.

AFAIU `Notify` is just like `watch::Receiver<_>` that doesn't remember the value, with some extra abilities to wake up one/specific waiting thread.

Reply:
~~i am positive you are wrong~~

yes you are correct